src/core/tactics.ml

```diff
@@ -85,7 +85,7 @@ let handle_tactic : Sig_state.t -> Proof.t -> p_tactic -> Proof.t =
       | Some gt -> Goal.get_type gt
       | None -> assert false
     in
-    let tt = Scope.scope_term Public ss env t in
+    let tt = Scope.scope_term Privat ss env t in
```

Comment:
```
constant symbol U : TYPE
private symbol V : U
definition W ≔ V
```

=> Private symbol not allowed here.

```
constant symbol U : TYPE
private symbol V : U
private definition W ≔ V
```

=> typechecks with master but needs here this modification in the scoping (as in this PR, definition uses also the proof mode)

``theorem`` begin opaque, it should be allowed to user any symbols it wants (disregarding the symbol exposition : public/private/...) in the proof script

But a ``definition`` being non-opaque, the scoping of the symbols used during the "proof" should depend on the exposition of the definition itself.

Possibilities for ``handle_tactic`` to know about the definition exposition :

- in sig_state
- new handle_refine parameter
- put it in data proof structure (in handle.ml)

Reply:
fixed in 90fc1e5